cmd/xl-storage.go

```diff
@@ -212,27 +212,22 @@ func newXLStorage(ep Endpoint) (s *xlStorage, err error) {
 		return nil, err
 	}
 
-	var rootDisk bool
-	if globalIsCICD {
-		rootDisk = true
-	} else {
-		rootDisk, err = disk.IsRootDisk(path, SlashSeparator)
+	rootDisk, err := disk.IsRootDisk(path, SlashSeparator)
+	if err != nil {
+		return nil, err
+	}
+	if !rootDisk && globalRootDiskThreshold > 0 {
+		// If for some reason we couldn't detect the root disk
+		// use - MINIO_ROOTDISK_THRESHOLD_SIZE to figure out if
+		// this disk is a root disk.
+		info, err := disk.GetInfo(path)
 		if err != nil {
 			return nil, err
 		}
-		if !rootDisk && globalRootDiskThreshold > 0 {
-			// If for some reason we couldn't detect the root disk
-			// use - MINIO_ROOTDISK_THRESHOLD_SIZE to figure out if
-			// this disk is a root disk.
-			info, err := disk.GetInfo(path)
-			if err != nil {
-				return nil, err
-			}
 
-			// treat those disks with size less than or equal to the
-			// threshold as rootDisks.
-			rootDisk = info.Total <= globalRootDiskThreshold
-		}
+		// treat those disks with size less than or equal to the
+		// threshold as rootDisks.
+		rootDisk = info.Total <= globalRootDiskThreshold
 	}
```

Comment:
removing this part won't work check there is a slight optimization as well when globalRootDiskThreshold is provided we don't have to detect RootDisk code at all we can directly do GetInfo()

PTAL #14239